### PR TITLE
rec: Stop using potentially offensive names internally and warn about deprecated settings.

### PR DESCRIPTION
**Pull request by omoerbeek**
### Short description

ATM table s used by both rec and auth, that seems a bit wrong, but otoh simplifies matters...

### Checklist

I have:
- [X] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [X] compiled this code
- [X] tested this code
- [ ] included documentation (including possible behaviour changes)
- [ ] documented the code
- [ ] added or modified regression test(s)
- [ ] added or modified unit test(s)
- [ ]  checked that this code was merged to master

### CODE REVIEW

File: pdns/pdns_recursor.cc [hunk 1]
```diff
@@ -4499,7 +4499,7 @@ static void setupNODThread()
   }
 }
 
-static void parseNODWhitelist(const std::string& wlist)
+static void parseNODIgnorelist(const std::string& wlist)
```
Comment (pieterlexis) [inline pdns/pdns_recursor.cc:4502]: a whitelist and an ignore list seem to be eachother's opposites?

File: pdns/pdns_recursor.cc [hunk 2]
```diff
@@ -5572,22 +5572,22 @@ int main(int argc, char **argv)
     ::arg().set("snmp-master-socket", "If set and snmp-agent is set, the socket to use to register to the SNMP daemon (deprecated)")="";
     ::arg().set("snmp-daemon-socket", "If set and snmp-agent is set, the socket to use to register to the SNMP daemon")="";
 
-    std::string defaultBlacklistedStats = "cache-bytes, packetcache-bytes, special-memory-usage";
+    std::string defeaultDisabledStats = "cache-bytes, packetcache-bytes, special-memory-usage";
```
Comment (pieterlexis) [inline pdns/pdns_recursor.cc:5575]: s/defeault/default/

File: pdns/pdns_recursor.cc (hunk 1 above)
Comment (omoerbeek) [reply pdns/pdns_recursor.cc:4502]: Original code used "whitelisted" to not report as newly observed. So this is consitent.

File: pdns/pdns_recursor.cc [hunk 3]
```diff
@@ -5057,15 +5057,15 @@ static int serviceMain(int argc, char*argv[])
 
   g_useKernelTimestamp = ::arg().mustDo("protobuf-use-kernel-timestamp");
 
-  blacklistStats(StatComponent::API, ::arg()["stats-api-blacklist"]);
-  blacklistStats(StatComponent::Carbon, ::arg()["stats-carbon-blacklist"]);
-  blacklistStats(StatComponent::RecControl, ::arg()["stats-rec-control-blacklist"]);
-  blacklistStats(StatComponent::SNMP, ::arg()["stats-snmp-blacklist"]);
+  disableStats(StatComponent::API, ::arg()["stats-api-blacklist"]);
```
Comment (pieterlexis) [inline pdns/pdns_recursor.cc:5060]: this still uses the old names of the configs. Perhaps we should merge their values into the new one?

Comment (omoerbeek) [reply pdns/pdns_recursor.cc:5060]: `disableStats` is both called for the old and new names, see below.

4.6.0 wil warn about them and 4.7 will remove the odl names.

Comment (pieterlexis) [reply pdns/pdns_recursor.cc:5060]: excellent

**Review by pieterlexis — APPROVED** [review]

**Pull request merged**